File: src/gui/src/profile/models/LegalStatusDto.ts
export default class LegalStatusDto {
    id: string;
    name: string;
    siret: string | null;
    vatNumber: string | null;
    vatExemption: boolean;
    vatRate: number | null;
    taxDeductionExemption: boolean;
    validFrom: string;
    validTo: string | null;

    public constructor(
        id: string,
        name: string,
        siret: string | null,
        vatNumber: string | null,
        vatExemption: boolean,
        vatRate: number | null,
        taxDeductionExemption: boolean,
        validFrom: string,
        validTo: string | null
    ) {
        this.id = id;
        this.name = name;
        this.siret = siret;
        this.vatNumber = vatNumber;
        this.vatExemption = vatExemption;
        this.vatRate = vatRate;
        this.taxDeductionExemption = taxDeductionExemption;
        this.validFrom = validFrom;
        this.validTo = validTo;
    }
}

File: src/backend/WordsmithHub.API/Features/LegalStatuses/Models/LegalStatusDto.cs
using WordsmithHub.Domain;

namespace WordsmithHub.API.Features.LegalStatuses.Models;

public record LegalStatusDto
{
    public required Guid Id { get; set; }
    public LegalStatusType? LegalStatusType { get; set; }
    public string? Siret { get; set; }
    public string? VatNumber { get; set; }
    public required bool VatExemption { get; set; }
    public decimal? VatRate { get; set; }
    public required bool TaxDeductionExemption { get; set; }
    public required DateTimeOffset ValidFrom { get; set; }
    public DateTimeOffset? ValidTo { get; set; }
}
